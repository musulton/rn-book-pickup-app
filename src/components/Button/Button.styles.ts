import {StyleSheet, TextStyle, ViewStyle} from "react-native";

import {COLORS, SIZES} from "../../constants";

interface Styles {
    container: ViewStyle
    text: TextStyle
}

const styles = StyleSheet.create<Styles>({
    container: {
        display: "flex",
        flex: 1,
        justifyContent: "center",
        flexDirection: "row",
        alignItems: "center",
        backgroundColor: COLORS.primary,
        borderRadius: SIZES.borderRadius
    },
    text: {
        padding: SIZES.padding1,
        fontWeight: "500",
        fontSize: 16,
        color: COLORS.white
    }
})

export default styles
