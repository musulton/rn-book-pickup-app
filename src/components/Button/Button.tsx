import React from "react";
import {Text, TouchableOpacity} from "react-native";

import styles from "./Button.styles";

interface ButtonProps {
    text: string
    onPress: () => void
}

const Button: React.FC<ButtonProps> = (props) => {
    return (
        <TouchableOpacity style={styles.container} onPress={props.onPress}>
            <Text style={styles.text}>{props.text}</Text>
        </TouchableOpacity>
    )
}

export default Button
