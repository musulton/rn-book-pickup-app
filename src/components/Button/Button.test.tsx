import {render, screen} from "@testing-library/react-native";

import Button from "./Button";

describe("<Button />", () => {
    it("should render Button correctly", () => {
        const props = {
            text: "Test",
            onPress: jest.fn(),
        }

        render(<Button {...props} />)

        expect(screen.toJSON()).toMatchSnapshot();
    })
});
