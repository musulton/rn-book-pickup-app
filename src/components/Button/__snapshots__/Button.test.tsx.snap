// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> should render Button correctly 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#039BE5",
      "borderRadius": 6,
      "display": "flex",
      "flex": 1,
      "flexDirection": "row",
      "justifyContent": "center",
      "opacity": 1,
    }
  }
>
  <Text
    style={
      {
        "color": "#FFFFFF",
        "fontSize": 16,
        "fontWeight": "500",
        "padding": 20,
      }
    }
  >
    Test
  </Text>
</View>
`;
